Covenant/Models/Listeners/BridgeListener.cs

```diff
@@ -184,7 +199,14 @@ private string NetworkReadString(NetworkStream stream, CancellationToken token)
                 readBytes = 0;
                 do
                 {
-                    readBytes = stream.Read(buffer, 0, buffer.Length);
+                    int currentBytesToRead = buffer.Length;
+                    int currentBytesLeft = len - totalReadBytes;
+                    if (currentBytesLeft < buffer.Length)
```

Comment:
If you use the "currentBytesToRead" variable, you should use it in the if statement, will be more clear.